src/app/App.js

```diff
@@ -15,22 +16,35 @@ import { usePhotos, useTodo } from 'shared/hooks';
 
 export const App = (props) => {
   const defaultCount = 4;
-  const photoState = usePhotos();
-  const todosState = useTodo();
+  const photoState = usePhotosStore();
+  const todosState = useTodosStore();
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    photoState.setPhotoCount(defaultCount);
-  }, []);
+    const loadData = async () => {
+      // Устанавливаем количество фотографий и задач
+      await Promise.all([
+        photoState.setPhotoCount(defaultCount),
+        todosState.setTodoCount(defaultCount),
+      ]);
 
-  useEffect(() => {
-    todosState.setTodoCount(defaultCount);
-  }, []);
+      setIsLoading(false); // Скрываем прелоудер после загрузки
+    };
+
+    loadData();
+  }, []); // Пустой массив зависимостей, чтобы эффект выполнялся только один раз при монтировании
 
   return (
     <div className={classes.app}>
-      <h3>{props.name}</h3>
-      <Gallery />
-      <Tasks />
+      {isLoading ? (
+        <Preloader />
+      ) : (
+        <>
+          <h3>{props.name}</h3>
+          <Gallery />
+          <Tasks />
+        </>
+      )}
```

Comment:
Если прелоудер будет иметь пропс, то ему можно будет передавать isLoading через этот пропс и это позволяет отказаться от условной конструкции - запись получится более аккуратной.
```suggestion
      <Preloader isActive={isLoading}/>
      <h3>{props.name}</h3>
      <Gallery />
      <Tasks />
```